File: PI_PERSO/WEB_SYMFONY/Reclamations/MobilePush/integration --finale--/WebTemplate/src/Entity/Event.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\EventRepository;

/**
 * Event
 *
 * @ORM\Table(name="event", indexes={@ORM\Index(name="host_fk", columns={"host_id"}), @ORM\Index(name="fk_locid", columns={"location_id"})})
 * @ORM\Entity(repositoryClass = "App\Repository\EventRepository")
 */
class Event
{
    /**
     * @var int
     *
     * @ORM\Column(name="event_id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $eventId;

    /**
     * @var string
     *
     * @ORM\Column(name="title", type="string", length=255, nullable=false)
     */
    private $title;

    /**
     * @var string
     *
     * @ORM\Column(name="type", type="string", length=255, nullable=false)
     */
    private $type;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="text", length=65535, nullable=false)
     */
    private $description;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="startDate", type="date", nullable=false)
     */
    private $startdate;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="endDate", type="date", nullable=false)
     */
    private $enddate;

    /**
     * @var int
     *
     * @ORM\Column(name="ticketCount", type="integer", nullable=false)
     */
    private $ticketcount;

    /**
     * @var string
     *
     * @ORM\Column(name="affiche", type="string", length=255, nullable=false)
     */
    private $affiche;

    /**
     * @var float
     *
     * @ORM\Column(name="ticketPrice", type="float", precision=10, scale=0, nullable=false)
     */
    private $ticketprice;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="host_id", referencedColumnName="id_user")
     * })
     */
    private $host;

    /**
     * @var \Location
     *
     * @ORM\ManyToOne(targetEntity="Location")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="location_id", referencedColumnName="num_loc")
     * })
     */
    private $location;

    

    // /**
    //  * Constructor
    //  */
    // public function __construct()
    // {
    //     $this->user = new \Doctrine\Common\Collections\ArrayCollection();
    // }

    public function getEventId(): ?int
    {
        return $this->eventId;
    }

    public function getTitle(): ?string
    {
        return $this->title;
    }

    public function setTitle(string $title): self
    {
        $this->title = $title;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getStartdate(): ?\DateTimeInterface
    {
        return $this->startdate;
    }

    public function setStartdate(\DateTimeInterface $startdate): self
    {
        $this->startdate = $startdate;

        return $this;
    }

    public function getEnddate(): ?\DateTimeInterface
    {
        return $this->enddate;
    }

    public function setEnddate(\DateTimeInterface $enddate): self
    {
        $this->enddate = $enddate;

        return $this;
    }

    public function getTicketcount(): ?int
    {
        return $this->ticketcount;
    }

    public function setTicketcount(int $ticketcount): self
    {
        $this->ticketcount = $ticketcount;

        return $this;
    }

    public function getAffiche(): ?string
    {
        return $this->affiche;
    }

    public function setAffiche(string $affiche): self
    {
        $this->affiche = $affiche;

        return $this;
    }

    public function getTicketprice(): ?float
    {
        return $this->ticketprice;
    }

    public function setTicketprice(float $ticketprice): self
    {
        $this->ticketprice = $ticketprice;

        return $this;
    }

    public function getHost(): ?User
    {
        return $this->host;
    }

    public function setHost(?User $host): self
    {
        $this->host = $host;

        return $this;
    }

    public function getLocation(): ?Location
    {
        return $this->location;
    }

    public function setLocation(?Location $location): self
    {
        $this->location = $location;

        return $this;
    }

    // /**
    //  * @return Collection<int, User>
    //  */
    // public function getUser(): Collection
    // {
    //     return $this->user;
    // }

    // public function addUser(User $user): self
    // {
    //     if (!$this->user->contains($user)) {
    //         $this->user->add($user);
    //     }

    //     return $this;
    // }

    // public function removeUser(User $user): self
    // {
    //     $this->user->removeElement($user);

    //     return $this;
    // }

}

File: PI_Integrated/Desktop Final Version/src/entities/Event.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package entities;

import java.sql.Date;

/**
 *
 * @author Aymen
 */
public class Event {
    private int event_id, ticketCount, host_id, location_id;
    private String title, type, description, affiche;
    private Date startDate, endDate;
    private float ticketPrice;

    public Event() {
    }

    public Event(int event_id) {
        this.event_id = event_id;
    }

    public Event(int event_id, int ticketCount, int host_id, int location_id, String title, String type, String description, Date startDate, Date endDate, String affiche) {
        this.event_id = event_id;
        this.ticketCount = ticketCount;
        this.host_id = host_id;
        this.location_id = location_id;
        this.title = title;
        this.type = type;
        this.description = description;
        this.startDate = startDate;
        this.endDate = endDate;
        this.affiche = affiche;
    }

    public Event(int ticketCount, int host_id, int location_id, String title, String type, String description, Date startDate, Date endDate, String affiche) {
        this.ticketCount = ticketCount;
        this.host_id = host_id;
        this.location_id = location_id;
        this.title = title;
        this.type = type;
        this.description = description;
        this.startDate = startDate;
        this.endDate = endDate;
        this.affiche = affiche;
    }

    public Event(int event_id, String title, String type) {
        this.event_id = event_id;
        this.title = title;
        this.type = type;
    }
    
    public int getEvent_id() {
        return event_id;
    }

    public void setEvent_id(int event_id) {
        this.event_id = event_id;
    }

    public int getTicketCount() {
        return ticketCount;
    }

    public void setTicketCount(int ticketCount) {
        this.ticketCount = ticketCount;
    }

    public int getHost_id() {
        return host_id;
    }

    public void setHost_id(int host_id) {
        this.host_id = host_id;
    }

    public int getLocation_id() {
        return location_id;
    }

    public void setLocation_id(int location_id) {
        this.location_id = location_id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getStartDate() {
        return startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public String getAffiche() {
        return affiche;
    }

    public void setAffiche(String affiche) {
        this.affiche = affiche;
    }
    
    public float getTicketPrice() {
        return ticketPrice;
    }

    public void setTicketPrice(float ticketPrice) {
        this.ticketPrice = ticketPrice;
    }

    @Override
    public String toString() {
        return "Event{" + "event_id=" + event_id + ", ticketCount=" + ticketCount + ", host_id=" + host_id + ", location_id=" + location_id + ", title=" + title + ", type=" + type + ", description=" + description + ", startDate=" + startDate + ", endDate=" + endDate + ", affiche=" + affiche + '}';
    }
    
    
}
